subsys/profiler/profiler_nordic.c

```diff
@@ -315,13 +324,19 @@ void profiler_log_add_mem_address(struct log_event_buf *buf,
 
 void profiler_log_send(struct log_event_buf *buf, uint16_t event_type_id)
 {
+	static uint32_t old_time;
+
 	__ASSERT_NO_MSG(event_type_id <= UCHAR_MAX);
 	if (sending_events) {
 		uint8_t type_id = event_type_id & UCHAR_MAX;
 
 		buf->payload_start[0] = type_id;
 		int key = irq_lock();
 
+		uint32_t new_time = k_cycle_get_32();
+
+		profiler_log_encode_LEB128(buf, (uint32_t) (new_time - old_time));
```

Comment:
Fix handling overflow of system counter (that could lead to strange issues). Cast is not needed, since both values are of the `uint32_t` type

Reply:
There is no overflow issue. Example: 1 - 0xFFFFFFFF = 0xFFFFFFFF00000002 => (uint32) => 2

Reply:
Ok. Sorry that I overlooked the previous comment